src/bindings/event.ts

```diff
@@ -0,0 +1,52 @@
+import { Context } from "probot";
+import { processors } from "../handlers/event";
+import { shouldSkip } from "../helpers";
+import { PayloadSchema } from "../types";
+import { ajv } from "../utils";
+
+let BotContext: Context = {} as Context;
+export const getBotContext = () => BotContext;
+
+export const bindEvents = async (context: Context): Promise<void> => {
+  BotContext = context;
+  const { payload, log, id, name } = context;
+
+  log.info(`Started binding events... id: ${id}, name: ${name}`);
+
+  const validate = ajv.compile(PayloadSchema);
+  const valid = validate(payload);
+  if (!valid) {
+    log.info("Payload schema validation failed!!!", payload);
+    log.error(validate.errors!);
+    return;
+  }
+  const { skip, reason } = shouldSkip();
+  if (skip) {
+    log.info(`Skipping the event. reason: ${reason}`);
+    return;
+  }
+
+  const handlers = processors[payload.action];
+  if (!handlers) {
+    log.error(`No handler configured for action: ${payload.action}`);
+    return;
+  }
+
+  const { pre, action, post } = handlers;
+  // Run pre-handlers
+  log.info(`Running pre handlers: ${pre.map((fn) => fn.name)}, action: ${payload.action}`);
+  for (const preAction of pre) {
+    await preAction();
+  }
+  // Run main handlers
+  log.info(`Running main handlers: ${action.map((fn) => fn.name)}, action: ${payload.action}`);
+  for (const mainAction of action) {
+    await mainAction();
+  }
+
+  // Run post-handlers
+  log.info(`Running post handlers: ${post.map((fn) => fn.name)}, action: ${payload.action}`);
+  for (const postAction of post) {
+    await postAction();
+  }
+};
```

Comment:
The reason why I thought about pre/post handlers is to process non event-specific actions for issues like unassign bounty hunters. 

PreActionHandler - it will be processed before main handler to do some. We can put a few of small functions into a prehandler list. In terms of completion time, it needs to be shorter one as much as possible. because we don't want for preActionhandlers to take much time. Most of time should definitely be spent by main handlers. we can just leave it empty if no need to use.  In terms of speed, it should be fast.
 
 
ActionHandler - it is the main callback function to do event-based things.
 
PostActionHandler - it would be processed after the main handlers.  We can put the functions into a posthandler list which will take a bit longer. If no needed, we can leave it empty. In terms of speed, it can be slow.

Reply:
> In terms of completion time, it needs to be shorter one as much as possible.

Isn't this negligible? I mean, like less than one second to do these actions? From what I understand (I could be wrong, my understanding of GitHub Actions Runners isn't master level) once it hits the JavaScript runtime, it runs at JavaScript speeds. Its just the Runner having to start its vps, load its OS, download the source code etc which takes most of the time.

Reply:
I think i didn't give you detailed meaning here. I just tried to emphasize the delay starting from gh event until main handler get started. Your understanding about Github Actions Runner is correct.